src/test/scala/org/apache/spark/sql/delta/MergeIntoScalaSuite.scala

```diff
@@ -19,14 +19,49 @@ package org.apache.spark.sql.delta
 import java.util.Locale
 
 import io.delta.tables._
-
 import org.apache.spark.sql._
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
 import org.apache.spark.sql.types.StructType
 
 class MergeIntoScalaSuite extends MergeIntoSuiteBase {
 
   import testImplicits._
 
+  test(s"test merge optimizeMatchedOnlyMerge enabled:true update") {
+    testMatchedOnly(true, true)
+  }
+  test(s"test merge optimizeMatchedOnlyMerge enabled:false update") {
```

Comment:
please follow the pattern of tests in MergeIntoSuiteBase for "insert only merge"
https://github.com/delta-io/delta/blob/master/src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala#L1585